enarx-keep-sev-shim/src/syscall.s

```diff
@@ -0,0 +1,53 @@
+# SPDX-License-Identifier: Apache-2.0
+
+.section .text, "ax"
+.global _syscall_enter
+.type _syscall_enter, @function
+.code64
+
+.p2align 4
+_syscall_enter:
+    swapgs                 # set gs segment to TSS
+    mov    %rsp,%gs:0x1c   # save userspace rsp
+    mov    %gs:0x4,%rsp    # load kernel rsp
+    pushq  $0x1b           # USER_DATA_SEGMENT
+    pushq  %gs:0x1c        # push userspace rsp
+    movq   $0x0,%gs:0x1c   # clear userspace rsp
+    push   %r11            # push rflags stored in r11
+    pushq  $0x23           # USER_CODE_SEGMENT
+    push   %rcx            # push userspace return pointer
+    swapgs                 # restore gs
+
+    # SYSV:    rdi, rsi, rdx, rcx, r8, r9
+    # SYSCALL: rdi, rsi, rdx, r10, r8, r9
+    mov    %r10, %rcx
+
+    # align stack
+    subq    $16, %rsp
```

Comment:
I don't understand how this aligns the stack. If the stack is unaligned, it will continue to be unaligned after you subtract 16.

Reply:
doh... it was 8 in the beginning... then s.th. changed and I added 8... lol